### DISCUSSION TITLE
mediamtx fail2ban filter

### DISCUSSION THREAD

**Discussion by adzmailru** [bluenviron/mediamtx#3487] [Questions]
### Question

Dear, Gurus!
Please help me!

I need to make fail2ban Log filter, part of "Regular expressions to match".

If you have, would you like to write it?

**Comment by aler9** [comment]
<s>Hello, in order to write a fail2ban filter, you can start by enabling debug logging inside MediaMTX:

```yml
logLevel: debug
```

The log lines you're interested into are these ones:

```
2024/08/27 23:06:55 DEB [RTSP] [conn 127.0.0.1:44480] [s->c] RTSP/1.0 401 Unauthorized
```

Therefore:

```
regex=".* DEB \[RTSP\] \[conn <HOST>:[0-9]+\] \[s->c\] RTSP/1.0 401 Unauthorized"
```

Then test it:

```
log_line="2024/08/27 23:06:55 DEB [RTSP] [conn 127.0.0.1:44480] [s->c] RTSP/1.0 401 Unauthorized"
regex=".* DEB \[RTSP\] \[conn <HOST>:[0-9]+\] \[s->c\] RTSP/1\.0 401 Unauthorized"
fail2ban-regex "$log_line" "$regex"
```
</s>

**Comment by aler9** [comment]
moved to https://github.com/bluenviron/mediamtx/issues/5016
